src/main/java/com/karankumar/bookproject/backend/util/PredefinedShelfUtils.java

```diff
@@ -56,6 +56,15 @@ public PredefinedShelfUtils(PredefinedShelfService predefinedShelfService) {
         }
     }
 
+    public PredefinedShelf findPredefinedShelf(String shelfName) {
+        return predefinedShelfService.findAllForLoggedInUser()
+                .stream()
+                .filter(shelf ->
+                        shelf.getPredefinedShelfName().equals(shelfName))
+                .collect(Collectors.toList())
+                .get(0); // there should only be one
+    }
+
```

Comment:
When I run the app and try to access a shelf (anything other than 'all books' from the dropdown list), this gives me an `IndexOutOfBoundsException: Index 0 out of bounds for length 0` -- it seems this isn't working as intended.

Could you please fix this and also write a test to verify this works?